dali/operators/random/rng_base_gpu.cuh

```diff
@@ -47,12 +48,13 @@ __device__ __inline__ void Generate(BlockDesc<false> desc,
   auto block_start = static_cast<T*>(desc.output);
   auto block_end = block_start + desc.size;
   for (auto out = block_start + threadIdx.x; out < block_end; out += blockDim.x) {
-    *out = ConvertSat<T>(dist(rng));
+    auto n = dist.Generate(rng);
+    *out = ConvertSat<T>(n);
   }
 }
 
-template <typename T, typename Dist, bool NeedsInput, bool DefaultDist>
-__global__ void RNGKernel(BlockDesc<NeedsInput>* __restrict__ block_descs,
+template <typename T, typename Dist, bool NoiseGen, bool DefaultDist>
```

Comment:
I'm not sure if `NoiseGen` is a good name. I would expect to pass the generator there, not true/false.
How about `ApplyDistribution`?

Reply:
I'll rename to IsNoiseGen